setup.py

```diff
@@ -225,12 +225,18 @@ def _find_include_file(self, include):
 
 
 def _find_library_file(self, library):
-    ret = self.compiler.find_library_file(self.compiler.library_dirs, library)
+    ret = self.compiler.find_library_file(
+        self.compiler.library_dirs, library, debug=debug_build()
+    )
     if ret:
         _dbg("Found library %s at %s", (library, ret))
-    else:
-        _dbg("Couldn't find library %s in %s", (library, self.compiler.library_dirs))
-    return ret
+        # we are only interested in the library name including a possible debug suffix
+        lib_name_base = os.path.basename(ret).split(".")[0]
+        # as the library prefix differs depending on library type and platform,
+        # we ignore it by looking for the actual library name
+        start_index = lib_name_base.find(library)
+        return lib_name_base[start_index:]
```

Comment:
This is a bit hacky, but I didn't find a better way to get the library name from the file name. I could not use `splitext`, as that would not handle something like `libfoo.so.6`. 
As this is an internal function only, it should be ok.